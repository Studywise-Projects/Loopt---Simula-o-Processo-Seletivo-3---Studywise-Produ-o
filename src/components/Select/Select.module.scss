@import '@/styles/_global.scss';

.select {
  height: 44px;
  text-align: left;
  width: 320px;

  @media only screen and (min-width: 1024px) {
    width: 420px;
  }
}

.menuItem {
  justify-content: flex-start;
}

.auxSelect {
  width: 240px;
}
