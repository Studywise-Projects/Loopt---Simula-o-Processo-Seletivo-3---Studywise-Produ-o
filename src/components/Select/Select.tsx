import {
  FormControl,
  InputLabel,
  MenuItem,
  Select as SelectMui,
} from '@mui/material';
import styles from './Select.module.scss';

interface ISelect {
  id: string;
  label: string;
  options: any;
  value: any;
  handleChange: any;
}

function Select({ id, label, options, value, handleChange }: ISelect) {
  return (
    <FormControl>
      <InputLabel id={id}>{label}</InputLabel>
      <SelectMui
        data-testid='select'
        id={id}
        label={label}
        value={value}
        onChange={handleChange}
        className={styles.select}
      >
        {options?.map((option: any, index: number) => (
          <MenuItem key={index} value={option} className={styles.menuItem}>
            {option.label ? option.label : option.name}
          </MenuItem>
        ))}
      </SelectMui>
    </FormControl>
  );
}

export default Select;
