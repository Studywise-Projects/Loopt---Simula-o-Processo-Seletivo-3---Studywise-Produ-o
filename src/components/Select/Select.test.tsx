import '@testing-library/jest-dom';
import { shallow, render } from 'enzyme';
import Select from './Select';

describe('Select', () => {
  it('Should match the snapshot', () => {
    const wrapper = shallow(
      <Select
        id='test-select'
        label='teste'
        options={undefined}
        value={undefined}
        handleChange={undefined}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('Should render correctly', () => {
    const wrapper = shallow(
      <Select
        id='test-select'
        label='teste'
        options={undefined}
        value={undefined}
        handleChange={undefined}
      />,
    );

    expect(wrapper.text()).toEqual('teste');
  });

  it('Should call handleChangeMock with the value = test', () => {
    const handleClickMock = jest.fn();

    const wrapper = shallow(
      <Select
        id='test-select'
        label='teste'
        options={undefined}
        value={undefined}
        handleChange={handleClickMock}
      />,
    );

    wrapper.find('[data-testid="select"]').simulate('change', 'Test');

    expect(handleClickMock).toBeCalledWith('Test');
  });
});
