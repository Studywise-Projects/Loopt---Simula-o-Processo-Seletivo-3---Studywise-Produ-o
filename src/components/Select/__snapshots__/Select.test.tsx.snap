// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Should match the snapshot 1`] = `ShallowWrapper {}`;
